### DISCUSSION TITLE
What's the difference between a plugin and an extension?

### DISCUSSION THREAD

**Discussion by g4bey** [tandemdude/hikari-lightbulb#247] [Q&A] [answered]
_What's the difference between a plugin and an extension?_

It appears to be similar, hence my question.

Would you register a plugin the same way you register an extension?

**Comment by tandemdude** [comment, marked as answer]
A plugin is effectively a "category" - it can have commands and listeners registered to it and can be plugged into and removed from the bot at any time.

An extension is a python file with a load function declared in it, and optionally an unload function, which can be loaded, unloaded and reloaded at any time. An extension can contain many plugins, commands, listeners, or anything else you may wish to have in the extension.
